src/attached_probe.cpp

```diff
@@ -713,7 +713,17 @@ void AttachedProbe::load_prog(BPFfeature &feature)
   std::string name;
   std::string tracing_type;
 
-  {
+  // Fallback attempts for kernels that don't cope well with func_infos and BTF
+  struct load_strategy {
+    bool load_btf;
+    bool func_infos;
+  } load_configs[] = {
+    { .load_btf = true, .func_infos = true },   // 1st, try both
+    { .load_btf = true, .func_infos = false },  // 2nd, without func_infos
+    { .load_btf = false, .func_infos = false }, // 3rd, without btf load as well
+  };
```

Comment:
I like this approach and I think that we should combine it with the loop for version attempts. Since kernel version checking was removed in https://github.com/torvalds/linux/commit/6c4fc209fcf9 (kernel version 5.0) and, IIUC, this PR is for 4.19, I think that the following sequence of attempts would make sense:
1. BTF + func_infos + no version
2. BTF + no func_infos + no version
3. no BTF + no func infos + version attempt 0
4. no BTF + no func infos + version attempt 1
5. no BTF + no func infos + version attempt 2

That should spare some `bpf_prog_load` calls and also make the semantics of individual attempts much clearer.

Reply:
Indeed. Done.

To make the version attempts obvious for the uninitiated, I named them using an enum, I hope you like it.

I hope you will like also logging of the retries:
I added a bit of logging for the retries and added a `std::stringstream` logbuf to append the info for each attempt into one line (and it is also needed to log inside the BTF/BTF load context as in it stderr is silenced by the silencer).

I guess that a logbuf for such situations may already exist, but I submit it now as-is. It helped me to confirm that the sequence is indeed executed as specified.

This example log is from different sequence, crafted to cover all added lines in this test sequence on the test host I used:
```
load tracepoint:raw_syscalls:sys_enter, with BTF, with func_infos: BTF load failed
load tracepoint:raw_syscalls:sys_enter, with BTF, version: 4.19.0: BTF load failed
load tracepoint:raw_syscalls:sys_enter, with func_infos = -7: Argument list too long
load tracepoint:raw_syscalls:sys_enter, version: 4.19.0: Success
```

As hinted at the end of the loop, I didn't de-indent the now context and indentation level that is now obsolete due to the removal of the inner kernel_version retriy loop:

I kept the context and indentation only to keep the commitdiff as small as possible. Of course it may be useful to de-indent that indentation level as a final step before or after the code review.